### ISSUE TITLE
Add database integrity checks for `ContentContainer` in the `space`module

### ISSUE THREAD

**Issue by marc-farre** [humhub/humhub#6900]
### Is your feature request related to a problem? Please describe.

If a space is deleted in the database, but for some reason, the related `ContentContainer` is not deleted, running a DB integrity check should delete it.

E.g: https://community.humhub.com/content/perma?id=276362

### Describe the solution you'd like

Add integrity checks in the `space` module similar to:
- https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/Events.php#L63-L70
- https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/Events.php#L144-L151

**Comment by luke-** [comment]
I can't find the check straight away either.

We should add a check here and loop over all "ContentContainers" and check whether the target is still correct. 

https://github.com/humhub/humhub/blob/master/protected/humhub/modules/content/Events.php#L93


@marc-farre What do you think?

**Comment by marc-farre** [comment]
Yes, I also thought about checking in the `content` module (this is where I searched first).
In this case, we can remove so checks in the `user` module.
